### ISSUE TITLE
fish shell: can't cross-compile

### ISSUE THREAD

**Issue by rembo10** [NixOS/nixpkgs#151978]
### Describe the bug
Trying to cross compile fish for armv7l-linux on x86_64-linux, I was running into an error where cmake was trying to use the host platform's uname (and gettext), instead of the build platform's versions. Checking the env-vars in the failed build directory, I saw that CMAKE_PREFIX_PATH was populated with all of the host platform's dependencies. From the [cmake docs:](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html), the CMAKE_PREFIX_PATH is a:

```
Semicolon-separated list of directories specifying installation prefixes to be searched by the find_package(),
find_program(), find_library(), find_file(), and find_path() commands. Each command will add appropriate
subdirectories (like bin, lib, or include) as specified in its own documentation.
```

Commenting out the addCMakeParams() in the cmake [setup-hook.sh](https://github.com/NixOS/nixpkgs/blob/ab93217a2b74a1c36bc892c14f44ee5959c33f12/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L2) fixes the issue, and all my other cmake dependencies seem to build fine (EDIT: other packages, e.g. aws-c-cal failed to build), but I'm not sure if this is the optimal solution, it's just something that happened to work.

It's a little above my pay grade and I don't really feel comfortable messing with something that might affect so many packages, so maybe someone can take a look? 

### Steps To Reproduce
1. nix build nixpkgs#pkgsCross.armv7l-hf-multiplatform.fish
(although maybe it's better to try with another cmake package since fish seems to bring in a whole nixos system? EDIT: although I made a [patch to fix that](https://github.com/NixOS/nixpkgs/pull/152025))

### Expected behavior
cmake should use the build host's binaries when building

### Additional context
I was able to cross compile fish on my machine at one point, but when I tried to build it in github actions, it was failing. I set sandbox = true in my nix.conf, and then I started getting the error on my machine as well (although now even with that setting disabled it still happens?)

I saw this in the failed build dir's CMakeSystem.cmake:
```
set(CMAKE_HOST_SYSTEM "Linux-/nix/store/gngmkdfmqz7plxckwdj3n5gyyai4nsqn-coreutils-armv7l-unknown-linux-gnueabihf-9.0/bin/uname: line 1: syntax error: unexpected word (expecting ")")")
set(CMAKE_HOST_SYSTEM_NAME "Linux")
set(CMAKE_HOST_SYSTEM_VERSION "/nix/store/gngmkdfmqz7plxckwdj3n5gyyai4nsqn-coreutils-armv7l-unknown-linux-gnueabihf-9.0/bin/uname: line 1: syntax error: unexpected word (expecting ")")")
set(CMAKE_HOST_SYSTEM_PROCESSOR "unknown")
```

### Notify maintainers
@ttuegel @LnL7

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.10, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.1`
```

**Comment by lifefinderglobal** [comment]
I

**Comment by cole-h** [comment]
Also cc @jtojnar -- I think you've done some CMake work in the past? Maybe you have any ideas?

**Comment by rembo10** [comment]
I think these flags need to be set when cross compiling:
set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)

**Comment by jiegec** [comment]
This issue is fixed by #152556.
